lisa/runner.py

```diff
@@ -33,6 +33,19 @@ def parse_testcase_filters(raw_filters: List[Any]) -> List[schema.BaseTestCaseFi
     return filters
 
 
+def print_results(
+    test_results: List[Union[TestResult, TestResultMessage]],
+    output_method: Callable[[str], None],
+    ) -> None:
+    for index, result in enumerate(test_results):
+        # Printing for TestResultMessage type.
+        if hasattr(result, "name"):
```

Comment:
You can use `isinstance` to check the type, like below. It should handle both TestResult and TestResultMessage well. And max the shared logic.

```python
output_method.info("________________________________________")
result_count_dict: Dict[TestStatus, int] = {}
for test_result in test_results:
    if isinstance(test_result, TestResult):
        result_name = test_result.runtime_data.metadata.full_name
        result_status = test_result.status
    elif isinstance(test_result, TestResultMessage):
        result_name = test_result.name
        result_status = test_result.status
    else:
        raise LisaException(f"Unknown result type: '{type(test_result)}'")
    output_method(
        f"{result_status.full_name:>50}: "
        f"{result_status.name:<8} {test_result.message}"
    )
    result_count = result_count_dict.get(result_status, 0)
    result_count += 1
    result_count_dict[result_status] = result_count
output_method.info("test result summary")
output_method.info(f"    TOTAL    : {len(test_results)}")
for key in TestStatus:
    count = result_count_dict.get(key, 0)
    if key == TestStatus.ATTEMPTED and count == 0:
        # attempted is confusing, if user don't know it.
        # so hide it, if there is no attempted cases.
        continue
    output_method.info(f"    {key.name:<9}: {count}")
```